### DISCUSSION TITLE
RTSP to VLC: CPU use increases constantly up to 100% after a bunch of hours

### DISCUSSION THREAD

**Discussion by vongillus** [bluenviron/mediamtx#1205] [Questions]
On Raspberry 4 with Raspbian Bullseye with the Raspberry Pi NoIR Camera Board v2 - 8 Megapixels.

Only RTSP is activated with following options:

paths:
  cam:
    source: rpiCamera
    rpiCameraWidth: 960
    rpiCameraHeight: 720
    rpiCameraFPS: 15

At the beginning the used CPU is like around 10%. After 3 hours it's already at 50% and if I wait longer it will at some point reach 100% CPU and after 24 hours the stream is not properly working.

Anyone having the same problem?

Thanks!

Here some details when it starts:

rtsp-simple-server[5815]: 2022/10/27 09:46:54 INF rtsp-simple-server v0.20.0
rtsp-simple-server[5815]: 2022/10/27 09:46:54 INF [path cam] [rpicamera source] started
rtsp-simple-server[5815]: 2022/10/27 09:46:54 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
rtsp-simple-server[5825]: [40:31:34.247510631] [5825]  INFO Camera camera_manager.cpp:293 libcamera v0.0.0+3406-e96d0201
rtsp-simple-server[5825]: [40:31:34.286555308] [5826]  WARN RPI raspberrypi.cpp:1211 Mismatch between Unicam and CamHelper for embedded data usage!
rtsp-simple-server[5825]: [40:31:34.287227781] [5826]  INFO RPI raspberrypi.cpp:1326 Registered camera /base/soc/i2c0mux/i2c@1/imx219@10 to Unicam device /dev/media0 and ISP device /dev/media2
rtsp-simple-server[5825]: [40:31:34.287982162] [5825]  INFO Camera camera.cpp:1028 configuring streams: (0) 960x720-YUV420
rtsp-simple-server[5825]: [40:31:34.288334083] [5826]  INFO RPI raspberrypi.cpp:747 Sensor: /base/soc/i2c0mux/i2c@1/imx219@10 - Selected sensor format: 1640x1232-SBGGR10_1X10 - Selected unicam format: 1640x1232-pBAA
rtsp-simple-server[5825]: [40:31:34.313656530] [5831]  INFO IPARPI raspberrypi.cpp:626 Request ctrl: FrameDurationLimits = [ 66666, 66666 ]
rtsp-simple-server[5815]: 2022/10/27 09:46:55 INF [path cam] [rpicamera source] ready: 1 track (H264)
rtsp-simple-server[5815]: 2022/10/27 09:47:11 INF [RTSP] [conn x.x.x.x:59839] opened
rtsp-simple-server[5815]: 2022/10/27 09:47:11 INF [RTSP] [session 514590450] created by x.x.x.x:59839
rtsp-simple-server[5815]: 2022/10/27 09:47:11 INF [RTSP] [session 514590450] is reading from path 'cam', with UDP, 1 track (H264)

**Comment by aler9** [comment]
Hello, does the CPU increase on the Rasberry Pi which is running rtsp-simple-server or on the pc which is running VLC?

[a comment by an automated account is not shown]
